tests/ClientTest.php

```diff
@@ -95,7 +95,7 @@ public function testSelectCollectionInheritsOptions(): void
         $this->assertInstanceOf(ReadConcern::class, $debug['readConcern']);
         $this->assertSame(ReadConcern::LOCAL, $debug['readConcern']->getLevel());
         $this->assertInstanceOf(ReadPreference::class, $debug['readPreference']);
-        $this->assertSame(ReadPreference::RP_SECONDARY_PREFERRED, $debug['readPreference']->getMode());
+        $this->assertSame(ReadPreference::SECONDARY_PREFERRED, $debug['readPreference']->getMode());
```

Comment:
This assert fails. The value is the int `6`, not the string `secondaryPreferred`. 
Looking at the [driver test](https://github.com/mongodb/mongo-php-driver/blob/master/tests/manager/bug0851-001.phpt#L42), the value of `Manager::getReadPreference()` should be the one sent in options?

Reply:
![image](https://github.com/mongodb/mongo-php-library/assets/400034/17779a02-74da-4d7b-8d2b-065433cc3940)

The inner value of `$debug['readPreference']->mode` is `secondaryPreferred`, but the method `getMode()` returns `6`.

Reply:
I could have used `get_object_vars` to access the inner property, but I'll compare `ReadPreference` objects instead.